horizon/middleware_manager.py

```diff
@@ -0,0 +1,144 @@
+import asyncio
+from copy import deepcopy
+import json
+from datetime import datetime
+
+from app.database import CREDENTIALS
+from horizon.middleware_api import get_players_online, authenticate_async, authenticate, get_active_games, get_recent_stats, get_recent_game_history
+from horizon.parsing.uya_game import uya_map_parser, uya_time_parser, uya_gamemode_parser
+
+from app.schemas.schemas import (
+    Pagination,
+    UyaPlayerOnlineSchema,
+    UyaGameOnlineSchema,
+    DeadlockedPlayerOnlineSchema,
+    DeadlockedGameOnlineSchema,
+)
+
+
+class MiddlewareManager:
```

Comment:
I don't think it's advantageous to use this as a base class for DL and UYA since the game has to be passed in as part of the child component anyway. There are a couple compositional mixin approaches that can be used as replacements for inheritance, but I think the simplest and best long-term option would be to duplicate the parent code for the Deadlocked and UYA child components. This will make the code more verbose, but easier to read, and it accounts for future changes to the individual APIs.